src/components/side-panel/property-inputs/ConfidenceProperties.tsx

```diff
@@ -0,0 +1,115 @@
+import { useContext, useEffect, useState } from "react";
+import "../../side-panel/side-panel.css";
+import { colors, dashStyles } from "../../../helper-functions/chart-helpers";
+import CustomSelect, { CustomSelectProps } from "../../select/CustomSelect";
+import ChartContext from "../../../context/ChartContext";
+import ColorOption from "../../select/ColorOption";
+
+interface Props {
+  activeSeries: string;
+  columnNames: string[];
+}
+
+const ConfidenceProperties = ({
+  activeSeries,
+  columnNames,
+}: Props): JSX.Element => {
+  const { selectedDimensions, setSelectedDimensions, chartProperties } =
+    useContext(ChartContext);
+  const [prevChartType, setPreviousChartType] = useState<any>("Line");
+  const [intervalType, setIntervalType] = useState("---");
+  const chartType = chartProperties?.chartTypes?.chartType;
+
+  useEffect(() => {
+    if (prevChartType === "Line" && intervalType === "intervals") {
+      updateAllDimensions("intervalType", "---");
+    }
+    setPreviousChartType(chartType);
+  }, [chartType]);
+
+  // Get the selected dimension based on the active series prop
+  const selectedDimension = selectedDimensions.find(
+    (d) => d.name === activeSeries,
+  );
+
+  const updateAllDimensions = (property: string, value: any) => {
+    setIntervalType(value);
+    setSelectedDimensions((prev) =>
+      prev.map((d) => ({ ...d, [property]: value })),
+    );
+  };
+
+  // Shared updater for updating the color and dashStyle properties on the selected dimension
+  const updateDimension = (property: string, value: any) => {
+    if (chartType !== "Line" && value === "intervals") {
+      alert("Confidence intervals are only available on line charts");
```

Comment:
Would it be significant work to make the dropdown more adaptive? So if the user has chosen Bar charts their only choices are 'None' or 'Error bars'? It avoids the user learning through getting it wrong, we just limit their choices to only the possible. Could be a separate PR/improvement for later if it'd be a big change

Reply:
I think that would be a good idea, but think it would be a fair amount of work and worth its own ticket.